app/queries/servers/custom_profile.ts

```diff
@@ -177,11 +177,8 @@ export const deleteCustomProfileAttributesByFieldId = async (database: Database,
     // Process attributes in batches to avoid performance issues with large datasets
     const promises = [];
     for (let i = 0; i < attributes.length; i += batchSize) {
-        const batch = attributes.slice(i, i + batchSize);
-        const preparedModels = batch.map((attribute) => attribute.prepareDestroyPermanently());
-
         const batchPromise = database.write(async () => {
-            await database.batch(...preparedModels);
+            await database.batch(...(attributes.slice(i, i + batchSize)));
```

Comment:
we should check how doing this globally in database.batch affects everything,.. also we should look into requesting an animation frame per batch operation

Reply:
@Willyfrog can we create a ticket for that ?